Serialization/DataSerializer.cs

```diff
@@ -0,0 +1,55 @@
+﻿using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Runtime.Serialization.Json;
+using System.Text;
+
+namespace SINEATER.Serialization
+{
+    public static class DataSerializer
+    {
+        public static T Load<T>(string json)
+        {
+            using (MemoryStream stream = new MemoryStream(Encoding.UTF8.GetBytes(json)))
+            {
+                DataContractJsonSerializer ser = new DataContractJsonSerializer(typeof(T), _types);
+                stream.Position = 0; // Ensure begining
+                T result = (T)ser.ReadObject(stream);
+                return result;
+            }
+        }
+
+        // Types that can be deserialized
+        private static List<Type> _types = new List<Type> {
```

Comment:
Use reflection here to extract everything that inherits from `ISkirmishStep`.

```
var types = AppDomain.CurrentDomain.GetAssemblies()
    .SelectMany(s => s.GetTypes())
    .Where(p => typeof(ISkirmishStep).IsAssignableFrom(p));
```

Reply:
Same for Trait, except I'm not sure that Trait currently has an interface, but add it if not - empty `ITrait` just for the purpose of reflection